### PR TITLE
Use exclude pattern when building dotnet dependency tree

### PR DESCRIPTION
**Pull request by attiasas**
- [x] The pull request is targeting the `dev` branch.
- [x] The code has been validated to compile successfully by running `go vet ./...`.
- [x] The code has been formatted properly using `go fmt ./...`.
- [x] All [static analysis checks](https://github.com/jfrog/jfrog-cli-security/actions/workflows/analysis.yml) passed.
- [x] All [tests](https://github.com/jfrog/jfrog-cli-security/actions/workflows/test.yml) have passed. If this feature is not already covered by the tests, new tests have been added.

-----

Use the exclude pattern to exclude sub-projects for `dotnet` when building its dependency tree.
Depends on:
https://github.com/jfrog/build-info-go/pull/233
https://github.com/jfrog/jfrog-cli-core/pull/1131

### CODE REVIEW

File: utils/auditbasicparams.go [hunk 1]
```diff
@@ -199,7 +205,26 @@ func (abp *AuditBasicParams) SetIsMavenDepTreeInstalled(isMavenDepTreeInstalled
...
+	abp.exclusions = exclusions
+	return abp
+}
+
+func (abp *AuditBasicParams) SetIsRecursiveScan(isRecursiveScan bool) *AuditBasicParams {
+	abp.isRecursiveScan = isRecursiveScan
+	return abp
+}
+
+func (abp *AuditBasicParams) IsRecursiveScan() bool {
```
Comment (eranturgeman) [inline utils/auditbasicparams.go:228]: any particular reason to change it to a function and not leaving it as a field?

**Review by eranturgeman — COMMENTED** [review]
some fixes suggested

File: commands/audit/scarunner.go [hunk 2]
```diff
@@ -108,13 +105,13 @@ func getRequestedDescriptors(params *AuditParams) map[coreutils.Technology][]str
...
 }
 
-func getExcludePattern(params *AuditParams, recursive bool) string {
-	exclusions := params.Exclusions()
-	if len(exclusions) == 0 {
-		exclusions = append(exclusions, DefaultExcludePatterns...)
-	}
-	return fspatterns.PrepareExcludePathPattern(exclusions, clientutils.WildCardPattern, recursive)
-}
+// func getExcludePattern(params *AuditParams) string {
```
[the remark this answers is not in the document]
Comment (attiasas) [reply commands/audit/scarunner.go]: done

File: commands/audit/audit.go [hunk 3]
```diff
@@ -96,9 +96,8 @@ func (auditCmd *AuditCommand) Run() (err error) {
 		SetMinSeverityFilter(auditCmd.minSeverityFilter).
 		SetFixableOnly(auditCmd.fixableOnly).
 		SetGraphBasicParams(auditCmd.AuditBasicParams).
-		SetThirdPartyApplicabilityScan(auditCmd.thirdPartyApplicabilityScan).
-		SetExclusions(auditCmd.exclusions).
-		SetIsRecursiveScan(isRecursiveScan)
+		SetThirdPartyApplicabilityScan(auditCmd.thirdPartyApplicabilityScan)
+	auditParams.SetIsRecursiveScan(isRecursiveScan).SetExclusions(auditCmd.Exclusions())
```
[the remark this answers is not in the document]
Comment (attiasas) [reply commands/audit/audit.go:100]: No, the parameters for the commands are a bit complicated.
What I do here is similar to what we do just above at: `SetGraphBasicParams(auditCmd.AuditBasicParams)`
We take the values stored at the command (provided by flags) and pass them to a new audit param used in the actual scanning and process.

**Pull request merged**